pkg/registry/pod/etcd/etcd_test.go

```diff
@@ -567,38 +530,40 @@ func TestEtcdCreateBinding(t *testing.T) {
 }
 
 func TestEtcdUpdateNotScheduled(t *testing.T) {
-	storage, _, _, fakeClient := newStorage(t)
+	storage, _, _, server := newStorage(t)
+	defer server.Terminate(t)
 	ctx := api.NewDefaultContext()
-	fakeClient.TestIndex = true
 
 	key, _ := storage.KeyFunc(ctx, "foo")
 	key = etcdtest.AddPrefix(key)
-	fakeClient.Set(key, runtime.EncodeOrDie(testapi.Default.Codec(), validNewPod()), 1)
+	if _, err := storage.Create(ctx, validNewPod()); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
 
 	podIn := validChangedPod()
 	_, _, err := storage.Update(ctx, podIn)
 	if err != nil {
 		t.Errorf("Unexpected error: %v", err)
 	}
-	response, err := fakeClient.Get(key, false, false)
+	obj, err := storage.Get(ctx, validNewPod().ObjectMeta.Name)
 	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
+		t.Errorf("unexpected error: %v", err)
 	}
-	podOut := &api.Pod{}
-	testapi.Default.Codec().DecodeInto([]byte(response.Node.Value), podOut)
-	if !api.Semantic.DeepEqual(podOut, podIn) {
+	podOut := obj.(*api.Pod)
+	// can not do full comparison b/c we are adding all the meta fields now
```

Comment:
Can you please explicitly say that validChangedPod() is changing only Labels, so we just compare this?